atlite/wind.py

```diff
@@ -90,3 +102,158 @@ def extrapolate_wind_speed(ds, to_height, from_height=None):
                           "units" : "m s**-1"})
 
     return wnd_spd.rename(to_name)
+
+
+# Cache
+_oedb_turbines = None
+
+def download_turbineconf(turbine, store_locally=True):
+    """Download a windturbine configuration from the OEDB database.
+    
+    Download the configuration of a windturbine model from the OEDB database
+    into the local 'windturbine_dir'.
+    The OEDB database can be viewed here:
+    https://openenergy-platform.org/dataedit/view/supply/turbine_library
+    (2019-07-22)
+    Only one turbine configuration is downloaded at a time, if the 
+    search parameters yield an ambigious result, no data is downloaded.
+    
+    Parameters
+    ----------
+    turbine : dict
+        Search parameters, either provide the turbine model id (takes priority)
+        or a manufacturer and turbine name, e.g. the following are identical:
+        {'id':10}, {'id':10, name:'E-53/800', 'manufacturer':'Unknown'},
+        {name:'E-53/800', 'manufacturer':'Enercon'}
+    store_locally : bool
+        (Default: True) Whether the downloaded config should be stored locally
+        in config.windturbine_dir.
+        
+    Returns
+    -------
+    turbineconf : dict
+        The turbine configuration downloaded and stored is also returned as a dict.
+        Has the same format as returned by 'atlite.ressource.get_turbineconf(name)'.
+    """
+
+    ## Parse information of different allowed 'turbine' values
+    parsed = False
+    # Assume id
+    if isinstance(turbine, int):
+        turbine = {'id': turbine}
+        parsed = True
+    elif isinstance(turbine,str):
+        s = turbine
+        turbine = {}
+        # Clean the string
+        s = s.strip()
+        s = s.replace('oedb:','')
+        
+        # 'turbine' is just a str(-inged) id
+        m = re.match("(^\d+$)", s)
+        if m:
+            turbine['id'] = turbine.get('id', int(m[1]))
+            s = "" # Consume the string after success
+            parsed = True
+
+        # 'turbine' is a name or combi of manufacturer + name
+        # Matches e.g. "TurbineName", "Manu1/Manu2[_ ]Turbine/with/number"
+        m = re.match("(?P<manufacturer>[-a-zA-Z0-9\/]*?)?(?:[\s_]*)?(?P<name>[-a-zA-Z0-9\/]+)$", s)
```

Comment:
I'm not a big fan of regular expressions, as they are always hard to read. Allowing to separate manufacturer and name with a space (`\s`) will break if one wants to use turbine names like:
```
download_turbineconf('E-126/4200 EP4')
download_turbineconf({'manufacturer': 'E-126/4200', 'name': 'EP4'})
```

Reply:
The thing is:

If this "convinience feature" allows for `_` to separate and optional manufacturer from a turbine name and turbine name and manufacturer can go ahead an contain special characters like `\s / . `, then the initial argument that "this is the way the file name will look like later" does not hold anymore (because that's what the file name will look like later).

And the `_` has then suddenly be degraded to become "just a separator".
Then we can also just use a real separator which is less commonly used in names, like `| ; ~ # `.
This brings us even further away from the "convinience feature".
If there is a turbine with spaces, slashes, long names, ... , then be all means: Please let us use the dictionary and the user has to input it accordingly.
Then everyone is happy: we don't need to assume much and the users gets what per is asking for (instead of what we make out of the request).